lib/harmony/resources/resources.go

```diff
@@ -0,0 +1,180 @@
+package resources
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	cl "github.com/Nv7-Github/go-cl"
+	ffi "github.com/filecoin-project/filecoin-ffi"
+	"github.com/filecoin-project/lotus/lib/harmony/harmonydb"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/pbnjay/memory"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/samber/lo"
+)
+
+var LOOKS_DEAD_TIMEOUT = 10 * time.Minute // Time w/o minute heartbeats
+
+type Resources struct {
+	Cpu       int
+	Gpu       float64
+	GpuRam    uint64
+	Ram       uint64
+	MachineID int
+}
+type Reg struct {
+	Resources
+	shutdown atomic.Bool
+}
+
+var logger = logging.Logger("harmonytask")
+
+var lotusRE = regexp.MustCompile("lotus-worker|lotus-harmony|yugabyted")
```

Comment:
Is yugabyted the single-node yugabyte binary?

* Do we actually want to disallow sharing one host between YB/lotus?
  * I get that one can run into resources of the other
  * But I'm not sure if it's worth complicating the setup for smaller SPs to the extent where you can't put YB next to one of lotus-providers
    * I can also see why we may want to push people to not de that
    * Maybe raise an Alert? (this would require setting up the Alerting system in lotus-provider, so.. idk, let's keep this constraint for now, will be easier to remove / loosen later than tighten it)
* In multi-node clusters the processes are called `yb-master` and `yb-tserver`

Reply:
It's less a constraint than a log complaint. 
I doubt that a small provider will overwhelm 1 reasonable machine with YB+Window+Winning, but I'd rather be extra chatty about what we aren't managing.